server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/model/MailboxFactory.scala

```diff
@@ -31,20 +31,16 @@ import reactor.core.scala.publisher.SMono
 import scala.jdk.CollectionConverters._
 import scala.jdk.OptionConverters._
 
-sealed trait MailboxConstructionOrder
-
-class Factory
-
 class MailboxFactory @Inject() (subscriptionManager: SubscriptionManager) {
 
   def create(mailboxMetaData: MailboxMetaData,
              mailboxSession: MailboxSession,
              allMailboxesMetadata: Seq[MailboxMetaData],
-             quotaLoader: QuotaLoader): SMono[Mailbox] = {
+             quotaLoader: QuotaLoader): SMono[Either[String, Mailbox]] = {
 
     val id: MailboxId = mailboxMetaData.getId
 
-    val name: MailboxName = MailboxName.liftOrThrow(mailboxMetaData.getPath
+    val name: Either[String, MailboxName] = MailboxName.validate(mailboxMetaData.getPath
```

Comment:
this part is ok: either the result of ```mailboxMetaData.getPath.getName.split(mailboxSession.getPathDelimiter).last``` is a valid MailboxName or not.
However, depending on where this value come from you'll have to transform error `String` into a domain error (very similar to exceptions actually and you may even throw it from here if convenient)